### PR TITLE
SOF Zephyr Power Management Architecture documentation

### PR DESCRIPTION
**Pull request by mwasko**

### CODE REVIEW

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
@mwasko wondering if there is any clever way we can populate state diagrams from doxygen/code. Would save keeping all aligned.
We should also says what version/platforms the features are supported from (included Zephyr version). @dbaluta I guess you are also NOT fully integrated with Zephyr PM yet ?

File: architectures/firmware/sof-zephyr/rtos_layer/power_management.rst [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+*************************************
+
+Zephyr is responsible for I/O devices power and clock management.
+
+The I/O device power is controlled based on usage count. More details can be
+found in `Zephyr Device Runtime Power Management
+documentation <https://docs.zephyrproject.org/latest/services/pm/device_runtime.html>`__
+
+The I/O clock gating is configurable in driver power policy. Each driver shall
+request desired clock and clock power gating if it is necessary for I/O,
```
Comment (deb-intel) [inline architectures/firmware/sof-zephyr/rtos_layer/power_management.rst:153]:
```suggestion
request the desired clock and clock power gating if it is necessary for I/O,
```

File: architectures/firmware/sof-zephyr/rtos_layer/power_management.rst [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+accelerator, etc. to work correctly.
+
+For instance, audio I/Os such as I2S associated with audio domain require high
```
Comment (deb-intel) [inline architectures/firmware/sof-zephyr/rtos_layer/power_management.rst:156]:
```suggestion
For instance, audio I/Os such as I2S associated with audio domain require a high
```

**Review by deb-intel — CHANGES_REQUESTED** [review]
These updates are simple grammatical updates to enhance the flow of your document. They can be merged "as is" into your PR.

**Comment by mwasko** [comment]
> These updates are simple grammatical updates to enhance the flow of your document. They can be merged "as is" into your PR.

@deb-intel thanks. Gramma updates merged and squashed into PR commit.

**Comment by deb-intel** [comment]
@mwasko Nice job! @lgirdwood are you ready for this to be merged? Do you want to wait on additional responses to earlier questions you had?

**Review by mmaka1 — APPROVED** [review]

**Pull request merged**